# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1266572667 -3600
#      Fri Feb 19 10:44:27 2010 +0100
# Node ID 21dd58bd683ca94a81391b0614cf1ab081128e81
# Parent  7c97da90fc8fe5b755162001c753f7ee3e6b0fb4
fix unsafe conversion in find

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-02-19  Jaroslav Hajek  <highegg@gmail.com>
+
+	* DLD-FUNCTIONS/find.cc (Ffind): Avoid unsafe conversion from Inf to
+	integer.
+
 2010-02-18  Jaroslav Hajek  <highegg@gmail.com>
 
 	* syscalls.cc (mk_stat_result): New helper function.
 	(Flstat): Call it here.
 	(Fstat): Also here. Handle also the fstat case here.
 
 2010-02-18  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
diff --git a/src/DLD-FUNCTIONS/find.cc b/src/DLD-FUNCTIONS/find.cc
--- a/src/DLD-FUNCTIONS/find.cc
+++ b/src/DLD-FUNCTIONS/find.cc
@@ -435,22 +435,22 @@ b = sparse(i, j, v, sz(1), sz(2));\n\
     }
 
   // Setup the default options.
   octave_idx_type n_to_find = -1;
   if (nargin > 1)
     {
       double val = args(1).scalar_value ();
 
-      if (error_state || (! xisinf (val) && (val < 0 || val != xround (val))))
+      if (error_state || (val < 0 || (! xisinf (val) && val != xround (val))))
         {
           error ("find: expecting second argument to be a nonnegative integer");
           return retval;
         }
-      else
+      else if (! xisinf (val))
         n_to_find = val;
     }
 
   // Direction to do the searching (1 == forward, -1 == reverse).
   int direction = 1;
   if (nargin > 2)
     {
       direction = 0;
